src/server.py

```diff
@@ -85,16 +86,19 @@ def _update_property_with_keyframes(self, element_name, control,
         if element is None:
             return
         current_value = element.get_property(prop_name)
-        control.unset_all()
         current_time = self.get_current_position()
         time_end = current_time + transition_time_ms * Gst.MSECOND
         self._add_keyframe_pair(control, current_time, current_value,
                                 time_end, prop_value)
 
-    def seek(self, position):
-        self.pipeline.seek_simple(Gst.Format.TIME,
-                                  Gst.SeekFlags.FLUSH | Gst.SeekFlags.KEY_UNIT,
-                                  position)
+    def seek(self, position=None, flags=None):
+        if flags is None:
+            flags = Gst.SeekFlags.FLUSH | Gst.SeekFlags.KEY_UNIT
```

Comment:
It would be better to make this a default parameter value in the function's signature.